lucene/core/src/java/org/apache/lucene/analysis/Analyzer.java

```diff
@@ -94,7 +94,7 @@
    * Create a new Analyzer, reusing the same set of components per-thread
    * across calls to {@link #tokenStream(String, Reader)}. 
    */
-  public Analyzer() {
```

Comment:
Can you not change those scopes in public API classes? This applies here and in other places -- protected changed to package-scope for source is not really an API-compatible change.

Reply:
I understand that it's notionally an API change, but `abstract` classes have no reason for public constructors. We can make everything protected and the subclasses that people use will be able to pick it up. I was over-zealous in a couple places going to package instead of protected, I'll fix that up.

Reply:
The constructors I understand, fine (although it's really a no-op change, as you indicated).